Bottom: 601b835a5d67a1d0e9d56fb0b23cfb46c058dd7b
Top:    877c05a5554d959b71adecb682d57da38cf54608
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-03-21 11:51:11 +0000

polygon.c: For some reason, the range computation in frac_circle appears to be wrong


---

diff --git a/src/polygon.c b/src/polygon.c
index b6a2807..4b2cacc 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -483,8 +483,8 @@ frac_circle (PLINE * c, Coord X, Coord Y, Vector v, int fraction)
   e1 = (v[0] - X) * POLY_CIRC_RADIUS_ADJ;
   e2 = (v[1] - Y) * POLY_CIRC_RADIUS_ADJ;
 
-  /* NB: the caller adds the last vertex, hence the -1 */
-  range = POLY_CIRC_SEGS / fraction - 1;
+  /* XXX */ /* NB: the caller adds the last vertex, hence the -1 */
+  range = POLY_CIRC_SEGS / fraction;
   for (i = 0; i < range; i++)
     {
       /* rotate the vector */
